.travis.yml

```diff
@@ -0,0 +1,20 @@
+language: scala
+jdk:
+  - openjdk8
+
+script:
+  - sbt ";+check ;+cpl ;+lint ;+doc ;+test"
+  - if [ -z "$TRAVIS_PULL_REQUEST" ] ; then ./sbt publish ; fi
```

Comment:
https://docs.travis-ci.com/user/environment-variables/

check `"${TRAVIS_REPO_SLUG}" == "scalaz/effect"`? because don't publish from forked repositories.